File: 이상욱/BOJ/step/BOJ1010.java
package boj;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Bridge {

	public double combination(double n, double r) {
		double a = 1, b = 1;
		for (double i = n; i > (n - r); i--) {
			a *= i;

		}
		for (double i = 2; i <= r; i++) {
			b *= i;
		}

		return a / b;

	}
}

public class BOJ1010 {
	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		int n = Integer.parseInt(st.nextToken());

		for (int i = 0; i < n; i++) {
			st = new StringTokenizer(br.readLine());

			double a = Integer.parseInt(st.nextToken());
			double b = Integer.parseInt(st.nextToken());
			Bridge brid = new Bridge();

			System.out.printf("%.0f", brid.combination(b, a));
			System.out.println();

		}
	}
}

File: 이상욱/BOJ/step/BOJ1010.kt
package level17

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

class Bridge{

    fun combination(n : Double, r : Double) : Double{

        var a = 1.0
        var b = 1.0

        for(i in n.toInt() downTo (n-r).toInt()+1){
            a *= i
        }

        for(i in 2 .. r.toInt()){
            b *= i
        }

        return a/b
    }

}

fun main(){

    val br = BufferedReader(InputStreamReader(System.`in`))
    var st = StringTokenizer(br.readLine())

    val n = st.nextToken().toInt()

    repeat(n){
        st = StringTokenizer(br.readLine())

        val a  = st.nextToken().toInt()
        val b = st.nextToken().toInt()

        val bridge = Bridge()

        print("%.0f".format(bridge.combination(b.toDouble(),a.toDouble())))
        println()
    }

}